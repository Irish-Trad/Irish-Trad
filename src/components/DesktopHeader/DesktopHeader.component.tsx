import React from 'react'
import Footer from 'components/Footer'
import Navbar from 'components/Navbar'

import { Header, LogoContainer, Image, Title } from './DesktopHeader.styles'

const DesktopHeader: React.FC = () => {
  return (
    <Header>
      <LogoContainer>
        <Image
          src="assets/high-quality/2243E.jpg"
          alt="Irish-Trad.net Logo"
          title="Irish-Trad.net Logo"
        />
        <Title>Irish-Trad.net</Title>
      </LogoContainer>
      <Navbar />
      <Footer />
    </Header>
  )
}

export default DesktopHeader
