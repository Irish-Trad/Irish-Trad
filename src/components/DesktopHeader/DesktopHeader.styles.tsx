import styled from 'styled-components'

export const Header = styled.header`
  height: 100vh;
  min-width: 300px;
  background-color: green;
  color: white;
  font-family: meathflfregular;
  font-size: 24px;
  flex-flow: column nowrap;

  display: none;

  @media screen and (min-width: 760px) {
    display: flex;
  }
`

export const LogoContainer = styled.div`
  display: flex;
  flex-flow: column nowrap;
  align-items: center;
  background: linear-gradient(to bottom, green, darkgreen);
  clip-path: ellipse(100% 100% at 50% 0%);
`

export const Image = styled.img`
  margin-top: 20px;
  width: 130px;
  height: 130px;
  border-radius: 50%;
  border: 3px solid white;
`

export const Title = styled.h1`
  font-size: 34px;
`
