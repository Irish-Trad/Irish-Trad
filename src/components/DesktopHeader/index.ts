export { default } from './DesktopHeader.component'
