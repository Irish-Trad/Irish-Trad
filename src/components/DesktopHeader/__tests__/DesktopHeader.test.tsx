import { render, cleanup } from '@testing-library/react'
import { HashRouter } from 'react-router-dom'
import DesktopHeader from '../DesktopHeader.component'

describe('DesktopHeader', () => {
  afterEach(() => {
    cleanup()
  })

  it('Snapshot', () => {
    const snapshot = render(
      <HashRouter>
        <DesktopHeader />
      </HashRouter>
    )

    expect(snapshot).toMatchSnapshot()
  })

  it('DesktopHeader should render all possible tabs', () => {
    const { getByText } = render(
      <HashRouter>
        <DesktopHeader />
      </HashRouter>
    )

    expect(getByText('Irish-Trad.net')).toBeInTheDocument()
    expect(getByText('Tablature')).toBeInTheDocument()
    expect(getByText('Lessons')).toBeInTheDocument()
    expect(getByText('Chords')).toBeInTheDocument()
    expect(getByText('Clef to Tab')).toBeInTheDocument()
    expect(getByText('Terminology')).toBeInTheDocument()
    expect(getByText('Community')).toBeInTheDocument()
    expect(getByText('Newsletter')).toBeInTheDocument()
  })
})
