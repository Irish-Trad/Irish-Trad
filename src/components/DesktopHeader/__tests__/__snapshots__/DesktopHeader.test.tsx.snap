// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DesktopHeader Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="sc-AxgMl cKCWWd"
      >
        <div
          class="sc-AxheI iEBshh"
        >
          <img
            alt="Irish-Trad.net Logo"
            class="sc-Axmtr fkWeGe"
            src="assets/high-quality/2243E.jpg"
            title="Irish-Trad.net Logo"
          />
          <h1
            class="sc-AxmLO oOKbw"
          >
            Irish-Trad.net
          </h1>
        </div>
        <nav
          class="sc-AxhCb cJSujL"
        >
          <a
            class="sc-AxhUy dokOxh"
            href="#/"
          >
            Home
          </a>
          <a
            class="sc-AxhUy dokOxh"
            href="#/tablature"
          >
            Tablature
          </a>
          <a
            class="sc-AxhUy dokOxh"
            href="#/lessons"
          >
            Lessons
          </a>
          <a
            class="sc-AxhUy dokOxh"
            href="#/chords"
          >
            Chords
          </a>
          <a
            class="sc-AxhUy dokOxh"
            href="#/clef-to-tab"
          >
            Clef to Tab
          </a>
          <a
            class="sc-AxhUy dokOxh"
            href="#/terminology"
          >
            Terminology
          </a>
          <a
            class="sc-AxhUy dokOxh"
            href="#/community"
          >
            Community
          </a>
          <a
            class="sc-AxhUy dokOxh"
            href="#/newsletter"
          >
            Newsletter
          </a>
        </nav>
        <footer
          class="sc-AxjAm fAuGiy"
        >
          <a
            class="sc-AxirZ ktnnZK"
            href="https://discord.gg/Mq9jeA8"
            target="_blank"
          >
            <i
              class="fa"
            />
          </a>
          <a
            class="sc-AxirZ ktnnZK"
            href="https://mailchi.mp/bfa9fa63aaab/irish-trad"
            target="_blank"
          >
            <i
              class="fa"
            />
          </a>
        </footer>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="sc-AxgMl cKCWWd"
    >
      <div
        class="sc-AxheI iEBshh"
      >
        <img
          alt="Irish-Trad.net Logo"
          class="sc-Axmtr fkWeGe"
          src="assets/high-quality/2243E.jpg"
          title="Irish-Trad.net Logo"
        />
        <h1
          class="sc-AxmLO oOKbw"
        >
          Irish-Trad.net
        </h1>
      </div>
      <nav
        class="sc-AxhCb cJSujL"
      >
        <a
          class="sc-AxhUy dokOxh"
          href="#/"
        >
          Home
        </a>
        <a
          class="sc-AxhUy dokOxh"
          href="#/tablature"
        >
          Tablature
        </a>
        <a
          class="sc-AxhUy dokOxh"
          href="#/lessons"
        >
          Lessons
        </a>
        <a
          class="sc-AxhUy dokOxh"
          href="#/chords"
        >
          Chords
        </a>
        <a
          class="sc-AxhUy dokOxh"
          href="#/clef-to-tab"
        >
          Clef to Tab
        </a>
        <a
          class="sc-AxhUy dokOxh"
          href="#/terminology"
        >
          Terminology
        </a>
        <a
          class="sc-AxhUy dokOxh"
          href="#/community"
        >
          Community
        </a>
        <a
          class="sc-AxhUy dokOxh"
          href="#/newsletter"
        >
          Newsletter
        </a>
      </nav>
      <footer
        class="sc-AxjAm fAuGiy"
      >
        <a
          class="sc-AxirZ ktnnZK"
          href="https://discord.gg/Mq9jeA8"
          target="_blank"
        >
          <i
            class="fa"
          />
        </a>
        <a
          class="sc-AxirZ ktnnZK"
          href="https://mailchi.mp/bfa9fa63aaab/irish-trad"
          target="_blank"
        >
          <i
            class="fa"
          />
        </a>
      </footer>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
